### PR TITLE
Issue 20: Fix date fields format after aggregation (D7 fix by morbiD)

### LINKED ISSUE #20: Date fields are not formatted after aggregation
When trying to use the "Maximum", "Minimum" function on any date field, e.g. the Post or Update date of nodes, it always renders as a timestamp, regardless of what date format is selected in the field configuration.

The reason seems to be that Views Aggregator Plus is treating the timestamp as a number and rendering it with the format_number() function.

This is a bug inherited from the Drupal 7 module, see the following issue: https://www.drupal.org/project/views_aggregator/issues/2728033

The patch provided there in [#7](https://www.drupal.org/project/views_aggregator/issues/2728033#comment-13042561) by morbiD seems to solve the problem. It can be also applied "as is" to the Backdrop module. I haven't reviewed the code yet, but I suggest to review and potentially integrate it into the Backdrop version of this module.

If needed, I can create a PR for this here.

### PR DESCRIPTION
**Pull request by AlexHoebart-ICPDR**
This fixes https://github.com/backdrop-contrib/views_aggregator/issues/20

Applied original D7 patch by morbiD from https://www.drupal.org/project/views_aggregator/issues/2728033#comment-13042561

### CODE REVIEW

File: views/views_aggregator_plugin_style_table.inc [hunk 1]
```diff
@@ -728,6 +735,64 @@ class views_aggregator_plugin_style_table extends views_plugin_style_table {
...
+   *
+   * @param object $field_handler
+   * @param int $value
+   *
+   * @return string, date formatted according to Views handler settings
+   *
+   * Note: this was taken in part from views_handler_field_date::render($values)
+   */
+  function render_date($field_handler, $value) {
+    $format = $field_handler->options['date_format'];
```
Comment (argiepiano) [inline views/views_aggregator_plugin_style_table.inc:753]: The following code was added to Backdrop's views_handler_field_date::render(), which is where the code for the method here came from. This should be added to your PR, please.

```suggestion
    if (!is_numeric($value)) {
      // If the value isn't numeric, assume it's an SQL DATETIME.
      $value = strtotime($value);
    }
    $format = $field_handler->options['date_format'];
```

File: views/views_aggregator_plugin_style_table.inc [hunk 2]
```diff
@@ -728,6 +735,64 @@ class views_aggregator_plugin_style_table extends views_plugin_style_table {
...
+    if (in_array($format, array('custom', 'raw time ago', 'time ago', 'today time ago', 'raw time hence', 'time hence', 'raw time span', 'time span', 'raw time span', 'inverse time span', 'time span'))) {
```
Comment (argiepiano) [inline views/views_aggregator_plugin_style_table.inc]: `'todat time ago'` has been removed from Backdrop. Please remove from here as well.

File: views/views_aggregator_plugin_style_table.inc [hunk 3]
```diff
@@ -728,6 +735,64 @@ class views_aggregator_plugin_style_table extends views_plugin_style_table {
...
+    }
+
+    $timezone = !empty($field_handler->options['timezone']) ? $field_handler->options['timezone'] : NULL;
+    $time_diff = REQUEST_TIME - $value; // will be positive for a datetime in the past (ago), and negative for a datetime in the future (hence)
+    switch ($format) {
+      case 'raw time ago':
+        return format_interval($time_diff, is_numeric($custom_format) ? $custom_format : 2);
+      case 'time ago':
+        return t('%time ago', array('%time' => format_interval($time_diff, is_numeric($custom_format) ? $custom_format : 2)));
+      case 'today time ago':
```
Comment (argiepiano) [inline views/views_aggregator_plugin_style_table.inc]: The format `'today time ago'` was removed from Backdrop. Please remove this `case` fully from your PR.

**Review by argiepiano — CHANGES_REQUESTED** [review]

**Comment by AlexHoebart-ICPDR** [comment]
Thank you very much for the review, I applied the changes.

**Pull request merged**